### DISCUSSION TITLE
Question on extending zimfw/zsh

### DISCUSSION THREAD

**Discussion by grzesuav** [zimfw/zimfw#429] [Q&A] [answered]
Hi,

firstly, thanks for awesome framework !

Actual question - is somewhere documented which repositories can be a zimfw modules ?  In https://zimfw.sh/docs/commands/ there is a note how to reference it (i.e. from github) - it can be any zsh repo or it need to follow some conventions ?

Cheers

**Comment by ericbn** [comment, marked as answer]
Hi @grzesuav. Actually there are some conventions to the module formats, based on how people create them in the "wild west out there". Basically there's the oh-my-zsh and the prezto formats, but they are not well documented anywhere. I'm considering documenting that, maybe in our wiki.

For now, you can infer the module format if you look at the "Initialization options" defaults in zmodule: https://zimfw.sh/docs/commands/#zmodule

EDIT: Also, I've briefly described the prezto module format, which is the one we use in out Zim modules, here: https://github.com/zimfw/zimfw/blob/master/CHANGELOG.md#143---2021-03-19

**Reply by grzesuav** [reply]
hi @ericbn , so let's say I want to add https://github.com/Aloxaf/fzf-tab as a module, I entered 
```shell
zmodule Aloxaf/fzf-tab 
```
and it worked like a charm :)

**Reply by grzesuav** [reply]
thank you !
